client/pages/Register.jsx

```diff
@@ -1,13 +1,33 @@
 import React, { useState } from 'react';
 import Button from '../components/Button';
-import { useForm } from 'react-hook-form';
+import { useForm, Controller } from 'react-hook-form';
+
 import { ErrorMessage } from '@hookform/error-message';
 const axios = require('axios');
 
 const Register = ({ onMount }) => {
   const [errorMessage, setErrorMessage] = useState(undefined);
 
-  const { register, handleSubmit, watch, formState: { errors } } = useForm();
+  const { control, register, handleSubmit, watch, formState: { errors } } = useForm();
+
+  const FileInput = ({ onChange }) => {
```

Comment:
React docs warn you to _never_ embed one component's code inside another component. Pull this out to the top level. Pass any props it might need.